Common/Systems/Synchronization/Handlers/SyncConditionalDropHandler.cs

```diff
@@ -8,33 +8,21 @@ namespace PathOfTerraria.Common.Systems.Synchronization.Handlers;
 /// </summary>
 internal class SyncConditionalDropHandler : Handler
 {
-	public override Networking.Message MessageType => Networking.Message.SyncConditionalDrop;
-
-	/// <inheritdoc cref="Networking.Message.SyncConditionalDrop"/>
-	public override void Send(params object[] parameters)
+	public static void Send(int id, bool add, int toClient = -1, int ignoreClient = -1)
 	{
-		CastParameters(parameters, out byte who, out int id, out bool add);
-
-		ModPacket packet = Networking.GetPacket(MessageType);
-		packet.Write(who);
+		ModPacket packet = Networking.GetPacket<SyncConditionalDropHandler>();
 		packet.Write(id);
 		packet.Write(add);
-		packet.Send();
-	}
-
-	internal override void ServerRecieve(BinaryReader reader)
-	{
-		SetValuesBasedOnReader(reader, true);
+		packet.Send(toClient, ignoreClient);
 	}
 
-	internal override void ClientRecieve(BinaryReader reader)
+	internal override void Receive(BinaryReader reader, byte sender)
 	{
-		SetValuesBasedOnReader(reader, false);
+		SetValuesBasedOnReader(reader, sender, Main.dedServ);
```

Comment:
Missing read on clients.
```suggestion
		byte who = sender >= byte.MaxValue ? reader.ReadByte() : sender;
		SetValuesBasedOnReader(reader, who, Main.dedServ);
```